xbmc/Application.cpp

```diff
@@ -1156,6 +1138,14 @@ bool CApplication::Initialize()
   g_curlInterface.Load();
   g_curlInterface.Unload();
 
+  if (g_windowManager.Initialized())
+  {
+    g_windowManager.CreateWindows();
+
+    if (g_advancedSettings.m_splashImage)
+      g_windowManager.ActivateWindow(WINDOW_SPLASH);
```

Comment:
move this 2 lines directly before https://github.com/xbmc/xbmc/pull/7650/files#diff-63246197ac0f0ac89358c1155ad21d5bR822 

`g_windowManager.Initialize()` will call `LoadNotOnDemandWindows()` which then holds the splash window.

```
g_windowManager.Add(new CGUIWindowSplash);
g_windowManager.Initialize();
```